tests/phpunit/includes/Core/Authentication/Clients/OAuth_Client_Empty.php

```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * OAuth_Client_Empty
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Authentication\Clients;
+
+use Google\Site_Kit\Core\Authentication\Clients\OAuth_Client_Base;
+
+class OAuth_Client_Empty extends OAuth_Client_Base {
```

Comment:
Why not use a mock instance instead? It essentially does the same thing just on-the-fly and gives us more control without the need for an added class just for tests.

Reply:
Hmm, I may not be aware of that possibility. Do we do that anywhere yet? How can I create a mock instance of an abstract class?

Reply:
Ah, it would be the same as creating a mock for any other class. PHPUnit creates a mock by creating a new class that extends the class it mocks so nothing special would be needed for an abstract class, so long as it didn't also have abstract methods – in which case I think the mock would just need to have those defined in the definition.

So I think we'd just need to `$this->getMockBuilder( OAuth_Client_Base::class )->getMock()`.  There's also a `createMock` method but it does a but more like disabling the constructor.